core/templates/pages/profile-page/profile-page.component.ts

```diff
@@ -16,218 +16,252 @@
-                UserService.getLoginUrlAsync().then(
-                  function(loginUrl) {
-                    if (loginUrl) {
-                      WindowRef.nativeWindow.location.href = loginUrl;
-                    } else {
-                      WindowRef.nativeWindow.location.reload();
-                    }
-                  }
-                );
-              } else {
-                if (!ctrl.isAlreadySubscribed) {
-                  ProfilePageBackendApiService.subscribeAsync(
-                    data.usernameOfViewedProfile
-                  ).then(() => {
-                    ctrl.isAlreadySubscribed = true;
-                    ctrl.updateSubscriptionButtonPopoverText();
-                    $scope.$apply();
-                  });
-                } else {
-                  ProfilePageBackendApiService.unsubscribeAsync(
-                    data.usernameOfViewedProfile
-                  ).then(() => {
-                    ctrl.isAlreadySubscribed = false;
-                    ctrl.updateSubscriptionButtonPopoverText();
-                    $scope.$apply();
-                  });
-                }
-              }
-            };
-
-            ctrl.updateSubscriptionButtonPopoverText = function() {
-              if (ctrl.userNotLoggedIn) {
-                ctrl.subscriptionButtonPopoverText = (
-                  'Log in or sign up to subscribe to your ' +
-                  'favorite creators.');
-              } else if (ctrl.isAlreadySubscribed) {
-                ctrl.subscriptionButtonPopoverText = (
-                  'Unsubscribe to stop receiving email notifications ' +
-                  'regarding new explorations published by ' +
-                  ctrl.username.value + '.');
-              } else {
-                ctrl.subscriptionButtonPopoverText = (
-                  'Receive email notifications, whenever ' +
-                  ctrl.username.value + ' publishes a new exploration.');
-              }
-            };
-            ctrl.updateSubscriptionButtonPopoverText();
 
-            ctrl.goToPreviousPage = function() {
-              if (ctrl.currentPageNumber === 0) {
-                $log.error('Error: cannot decrement page');
-              } else {
-                ctrl.currentPageNumber--;
-                ctrl.startingExplorationNumber = (
-                  ctrl.currentPageNumber * ctrl.PAGE_SIZE + 1
-                );
-                ctrl.endingExplorationNumber = (
-                  (ctrl.currentPageNumber + 1) * ctrl.PAGE_SIZE
-                );
-              }
-            };
-            ctrl.goToNextPage = function() {
-              if ((ctrl.currentPageNumber + 1) * ctrl.PAGE_SIZE >= (
-                data.editedExpSummaries.length)) {
-                $log.error('Error: Cannot increment page');
+interface Username {
+  title: string,
+  value: string,
+  helpText: string
+}
+
+interface UserDisplayedStatistic {
+  title: string,
+  value: number,
+  helpText?: string
```

Comment:
Please add an explanation of why this can be optional.

Reply:
I made this optional due to this assignment from the old code.
![image](https://user-images.githubusercontent.com/24855641/121226066-c713f680-c8a7-11eb-8a36-fefa7c1659c5.png)

Reply:
Okay, please add a comment about this and modify it in a way that you pass `null` explicitly. The type will then be `helpText: string | null`.

Reply:
done